test_restify fails with Python 3.11a7

### Describe the bug

tests/test_util_typing.py fails with Python 3.11a7 with the traceback:

```
    def test_restify():
        assert restify(int) == ":py:class:`int`"
        assert restify(int, "smart") == ":py:class:`int`"
    
        assert restify(str) == ":py:class:`str`"
        assert restify(str, "smart") == ":py:class:`str`"
    
        assert restify(None) == ":py:obj:`None`"
        assert restify(None, "smart") == ":py:obj:`None`"
    
        assert restify(Integral) == ":py:class:`numbers.Integral`"
        assert restify(Integral, "smart") == ":py:class:`~numbers.Integral`"
    
        assert restify(Struct) == ":py:class:`struct.Struct`"
        assert restify(Struct, "smart") == ":py:class:`~struct.Struct`"
    
        assert restify(TracebackType) == ":py:class:`types.TracebackType`"
        assert restify(TracebackType, "smart") == ":py:class:`~types.TracebackType`"
    
>       assert restify(Any) == ":py:obj:`~typing.Any`"
E       AssertionError: assert ':py:class:`~typing.Any`' == ':py:obj:`~typing.Any`'
E         - :py:obj:`~typing.Any`
E         ?     ^^^
E         + :py:class:`~typing.Any`
E         ?     ^^^^^

tests/test_util_typing.py:55: AssertionError
=========================== short test summary info ============================
FAILED tests/test_util_typing.py::test_restify - AssertionError: assert ':py:...
```

### How to Reproduce

Run the test suite against Python 3.11a7


### Expected behavior

_No response_

### Your project

-

### Screenshots

_No response_

### OS

Fedora Linux 37

### Python version

Python 3.11a7

### Sphinx version

4.5.0

### Sphinx extensions

_No response_

### Extra tools

_No response_

### Additional context

_No response_

--- a/sphinx/util/typing.py
+++ b/sphinx/util/typing.py
@@ -215,6 +215,9 @@ def _restify_py37(cls: Optional[Type], mode: str = 'fully-qualified-except-typin
         return text
     elif isinstance(cls, typing._SpecialForm):
         return ':py:obj:`~%s.%s`' % (cls.__module__, cls._name)
+    elif sys.version_info >= (3, 11) and cls is typing.Any:
+        # handle bpo-46998
+        return f':py:obj:`~{cls.__module__}.{cls.__name__}`'
     elif hasattr(cls, '__qualname__'):
         if cls.__module__ == 'typing':
             return ':py:class:`~%s.%s`' % (cls.__module__, cls.__qualname__)

